### DISCUSSION TITLE
child process stdio problem

### DISCUSSION THREAD

**Discussion by ObscurusGrassator** [nodejs/help#5029] [Q&A]
### Version

18.13.0

### Platform

Darwin ALFA.home 21.6.0 Darwin Kernel Version 21.6.0: Sun Nov  6 23:31:16 PST 2022; root:xnu-8020.240.14~1/RELEASE_X86_64 x86_64

### What steps will reproduce the bug?

https://github.com/nodejs/node/issues/new?assignees=&labels=&template=1-bug-report.yml

I don't know exactly what the problem is, but since no one knows the solution, I assume it is a bug. I'll try to describe it in the examples.

I can't catch the chrome child process output without closing the stream.

uncork() not work:
```js
const { spawn } = require('child_process');

let chrome = spawn(
    '/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome',
    [
        '--new-window', '--window-size=1200,1000',
        '--headless', '--disable-gpu', '--repl',
        'https://www.google.com'
    ],
);
chrome.stdout.on('data', data => { console.log('stdout:', data.toString('utf8')); });
chrome.stderr.on('error', err => { console.warn('stderr:', err.toString('utf8')); });
chrome.on('error',        err => { console.error(err.toString('utf8')); });
chrome.on('exit',        code => { console.log('Exit code:', code); });

setTimeout(() => {
    chrome.stdin.cork();
    chrome.stdin.write('123' + '\n', 'utf8', err => err && console.error(err));
    chrome.stdin.uncork(); // TODO not work - not send data to stream

    // chrome.stdin.end(); // this work, but it closes stream
}, 1000);

// missed console.log('stdout:' ...
```

stdout.on('data') not work:
```js
const { spawn } = require('child_process');

let chrome = spawn(
    '/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome',
    [
        '--new-window', '--window-size=1200,1000',
        '--headless', '--disable-gpu', '--repl',
        'https://www.google.com'
    ], {
        stdio: ['pipe', process.stdout, process.stderr]
    }
);

// TODO not work - output is showed in terminal output, but event is not called
process.stdout.on('data', data => { console.log('stdout:', data.toString('utf8')); });
process.stderr.on('error', err => { console.warn('stderr:', err.toString('utf8')); });

chrome.on('error',        err => { console.error(err.toString('utf8')); });
chrome.on('exit',        code => { console.log('Exit code:', code); });

process.stdin.write('111' + '\n', 'utf8', err => err && console.error(err));

setTimeout(() => {
    process.stdin.write('222' + '\n', 'utf8', err => err && console.error(err));
}, 1000);

// missed console.log('stdout:' ...
```

### What is the expected behavior?

I need to work from stdin and stdout without closing the stream.

### What do you see instead?

...

**Comment by mscdex** [comment]
Have you verified this is a problem with node and not with how the child process operates normally?

**Comment by ObscurusGrassator** [comment]
I wrote in the description that I don't know where the problem is. But with the bash console command itself, it is possible to communicate without any problems. 
```sh
/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome \
--headless --disable-gpu --repl https://www.google.com
```
Could you please see where the problem is please? And if it's not a bug, write, there is a problem, and what are the solutions?

**Comment by bnoordhuis** [comment]
I'll convert this to a discussion because it's unlikely to be an issue with node itself. More likely chrome detects what type of file descriptors its stdio are and adjusts accordingly.

**Reply by ObscurusGrassator** [reply]
If Node.js cannot run the command in the terminal environment (simulate the same environment), then I think it is a big problem. At least from my lay perspective.

**Reply by bnoordhuis** [reply]
You're of course entitled to your opinions. There's an open issue (https://github.com/nodejs/node/issues/46105) to implement pseudo-tty support that will probably help with that but don't expect to see it arrive anytime soon.
